src/emmc.c

```diff
@@ -0,0 +1,190 @@
+#include <errno.h>
+#include <fcntl.h>
+#include <glib/gstdio.h>
+#include <linux/major.h>
+#include <linux/mmc/ioctl.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+#include "emmc.h"
+#include "update_handler.h"
+
+
+static int r_emmc_read_extcsd(int fd, __u8 *extcsd) {
+	struct mmc_ioc_cmd cmd;
+
+	memset(&cmd, 0, sizeof(cmd));
+	memset(extcsd, 0, sizeof(__u8) * 512);
+
+	cmd.write_flag = 0;
+	cmd.opcode = MMC_SEND_EXT_CSD;
+	cmd.arg = 0;
+	cmd.flags = MMC_RSP_SPI_R1 | MMC_RSP_R1 | MMC_CMD_ADTC;
+	cmd.blksz = 512;
+	cmd.blocks = 1;
+	mmc_ioc_cmd_set_data(cmd, extcsd);
+
+	return ioctl(fd, MMC_IOC_CMD, &cmd);
+}
+
+gboolean r_emmc_read_bootpart(const gchar *device, gint *bootpart_active, GError **error) {
+	int ret;
+	gboolean res = FALSE;
+	__u8 extcsd[512];
+	int fd;
+	/* count from 1 */
+	int active_partition = -1;
+
+	fd = open(device, O_RDONLY);
+
+	if (fd == -1) {
+		g_set_error(error, R_UPDATE_ERROR, R_UPDATE_ERROR_FAILED,
+				"opening output device failed: %s", strerror(errno));
```

Comment:
use `g_strerror()` here

Reply:
Mh, okay. `strerror()` is used in other places too, e.g. [here](https://github.com/rauc/rauc/blob/master/src/update_handler.c#L41) and [here](https://github.com/rauc/rauc/blob/master/src/update_handler.c#L68). More patch opportunities :)